src/each.js

```diff
@@ -24,21 +23,26 @@ const iterateObject = (obj, fn) => {
 }
 
 /**
- * Takes a an array or object
+ * Takes an array or object
  * and a function, and runs the function
  * on each element
  * @param {array|object} list
  * @param {function} fn
  * @returns {any}
  * @example
  * each([ 'a', 'b', 'c' ], id) // => 'a'
- * each() // => undefined
+ * each() // => void
  */
 
-// $FlowFixMe
-const each = (list, fn) =>
-  isArrayLike(list)
-    ? iterateArray(list, fn)
-    : iterateObject(list, fn)
+const each = <T>(list: ?(*[] | {[string]: *}), fn: ?((?*, (number | string)) => T)) => {
+  if (list && fn) {
+    if (Array.isArray(list)
+    ) {
```

Comment:
This should be isArrayLike; right now `each` and `filter` also work for strings.